### PR TITLE
Destroy newly created aggregate device if its initialization fails

### PR DESCRIPTION
**Pull request by ChunMinChang**
If the created aggregate device (via `create_blank_device_sync`) fails during its initialization (e.g., `set_sub_devices_sync`, `set_master_device`, ...), the newly created aggregate device should be destroyed.

https://github.com/ChunMinChang/cubeb-coreaudio-rs/blob/91a7bb50198a9b7578d98d56e9fd32f257abb41f/src/backend/aggregate_device.rs#L37-L42

### CODE REVIEW

**Comment by ChunMinChang** [comment]
The test fails until #111 is merged.

File: src/backend/utils.rs [hunk 1]
```diff
@@ -35,18 +35,32 @@ pub fn cubeb_sample_size(format: fmt) -> usize {
...
         }
     }
 }
 
-pub fn finally<F: FnOnce()>(f: F) -> impl Drop {
-    Finalizer(Some(f))
+pub fn finally<F: FnOnce()>(f: F) -> Finalizer<F> {
+    Finalizer {
+        func: Some(f),
+        commit: true,
```
Comment (padenot) [inline src/backend/utils.rs]: I'd have merged `commit` and `func`. If `func` is `Some` then run it. Otherwise, do nothing.

**Review by padenot — APPROVED** [review]
r+ with a suggestion if you want.

File: src/backend/utils.rs (hunk 1 above)
Comment (ChunMinChang) [reply src/backend/utils.rs]: Yeah, that's neat! I'll update the patch! Thanks!

**Pull request merged**